plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3342,4 +3348,59 @@ public void testDroppingIcebergAndCreatingANewTableWithTheSameNameShouldBePossib
         assertThat(query("SELECT min(a_varchar) FROM test_iceberg_recreate")).matches("VALUES CAST('Trino' AS varchar)");
         dropTable("test_iceberg_recreate");
     }
+
+    @Test
+    public void testPathHiddenColumn()
+    {
+        String tableName = "test_path_" + randomTableSuffix();
+        @Language("SQL") String createTable = "CREATE TABLE " + tableName + " " +
+                "WITH ( partitioning = ARRAY['col1'] ) AS " +
+                "SELECT * FROM (VALUES " +
+                "(0, 0), (3, 0), (6, 0), " +
+                "(1, 1), (4, 1), (7, 1), " +
+                "(2, 2), (5, 2) " +
+                " ) t(col0, col1) ";
+        assertUpdate(createTable, 8);
+        assertTrue(getQueryRunner().tableExists(getSession(), tableName));
+
+        TableMetadata tableMetadata = getTableMetadata(ICEBERG_CATALOG, TPCH_SCHEMA, tableName);
+
+        List<String> columnNames = ImmutableList.of("col0", "col1", PATH_COLUMN_NAME);
+        List<ColumnMetadata> columnMetadatas = tableMetadata.getColumns();
+        assertEquals(columnMetadatas.size(), columnNames.size());
+        for (int i = 0; i < columnMetadatas.size(); i++) {
+            ColumnMetadata columnMetadata = columnMetadatas.get(i);
+            assertEquals(columnMetadata.getName(), columnNames.get(i));
+            if (columnMetadata.getName().equals(PATH_COLUMN_NAME)) {
+                // $path should be hidden column
+                assertTrue(columnMetadata.isHidden());
+            }
+        }
```

Comment:
You can test this more easily inspecting the result of `DESCRIBE table`.

Reply:
This is actually not dependent on https://github.com/trinodb/trino/pull/11661#discussion_r844185392.
I believe you can simplify both places. Note: i am asking to simplify, not to remove test coverage. 
Inspecting `DESCRIBE` output is sufficient to verify the `$path` column isn't there (it's hidden, or does not exist).
The selecting it is sufficient to verify it exists.

Reply:
@findepi edited the test method to keep it simple.